### DISCUSSION TITLE
Is it possible to drag items between groups in interactive mode?

### DISCUSSION THREAD

**Discussion by mkaatman** [Lissy93/dashy#533] [Ideas]
When interactive mode is on I find myself trying to drag and drop items between groups. Has this been discussed previously?

**Comment by lissy93** [comment]
Yup, you can move items. When in edit mode, you can right-click on any item to copy, move, duplicate or delete it.

<p align="center"><img width="500" src="https://i.ibb.co/sgRzt73/right-click-menu.png" /></p>
<p align="center"><img width="500" src="https://i.ibb.co/p2Xd5Nm/copy-move-item.png" /></p>

If you prefer the JSON editor, it's also possible to drag-to-rearrange objects there, as well as duplicate, insert, remove and edit items.
<p align="center"><img  width="200" src="https://user-images.githubusercontent.com/1862727/147097647-d8ccb5a5-37ac-4625-9bf2-644a4dbc731a.png" /></p>

---

In terms or dragging + dropping, that is on the [Roadmap](https://github.com/Lissy93/dashy/discussions/405).
I found it trickier than expected to do without using a third-party library, which is why it was put onto the backlog, but it should be possible.
But if you would like to try implementing this, that would be awesome.

**Reply by mkaatman** [reply]
Okay great. I didn't think to check for a context menu.

I haven't checked yet but is it using react?

**Reply by lissy93** [reply]
No problem :)
And it's Vue

**Comment by mkaatman** [comment]
Hey @Lissy93 one more question for you. I'm realizing that my config is not saving to disk despite a success confirmation.

Any change at all goes away with a browser refresh. Any thoughts on where I should start troubleshooting?

I saw it mentioned in the FAQ.
https://dashy.to/docs/troubleshooting#config-not-updating

I manually cleared local storage to make sure that wasn't the problem, it seems to be this second problem, but that shouldn't happen if I'm editing through the UI correct?

Thanks.

**Reply by lissy93** [reply]
Hmm, that's weird. Is this a new issue, or did this never work for you? if you [open up the browser console](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#how-to-open-browser-console), and repeat the action, do you see any errors? And if you open the app in incognito mode / private window, does the issue persist? And just to check, if you look at the conf.yml file, you don't see any of your changes?

**Reply by lissy93** [reply]
Actually I managed to recreate this, so I think it's a bug, looking into it at the moment...
Weirdly it works fine when running with Docker for me, how are you running the app?

**Reply by mkaatman** [reply]
No errors in developer console that I see. Same problem in incognito.

This is a new setup. 

I used tteck's script to set it up which isn't using docker:
https://raw.githubusercontent.com/tteck/Proxmox/main/setup/dashy-install.sh

Are you referring to the conf.yml in dist? I don't see changes being written and if I manually edit that file, I don't see them reflected in dashy.

**Reply by tteck** [reply]
App is built from source. After getting everything setup the way you want in interactive mode and saved to disk, you have to go into update configuration and rebuild application.

**Comment by mkaatman** [comment]
That was it. Thanks @tteck! Now I have more feedback! 

Because I was having issues I was overwriting my config repeatedly without realizing it. Is there any way to do a diff between the running config and the config saved to disk with maybe a reminder to manually rebuild?

Also on update config where it shows "Update Available" would it be possible to make that a link to the changelog?
